crates/storage/provider/src/providers/snapshot/writer.rs

```diff
@@ -216,6 +217,26 @@ impl<'a> SnapshotProviderRW<'a> {
         Ok(self.writer.user_header().tx_end())
     }
 
+    /// Appends header to snapshot file.
+    ///
+    /// Returns the current [`BlockNumber`] as seen in the static file.
+    pub fn append_header(
+        &mut self,
+        hash: BlockHash,
+        header: Header,
+        terminal_difficulty: U256,
+    ) -> ProviderResult<BlockNumber> {
+        debug_assert!(self.writer.user_header().segment() == SnapshotSegment::Headers);
+
+        self.writer.user_header_mut().increment_block();
+
+        self.append_column(hash)?;
+        self.append_column(header)?;
+        self.append_column(terminal_difficulty)?;
```

Comment:
order is:
`header`, `td` , `hash` 

according to `crates/storage/db/src/snapshot/masks.rs`:

```rust
add_snapshot_mask!(HeaderMask, Header, 0b001);
add_snapshot_mask!(HeaderMask, <HeaderTD as Table>::Value, 0b010);
add_snapshot_mask!(HeaderMask, BlockHash, 0b100);
```